File: src/app/services/weather.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { EntropyReading, EntropyHistory } from '../models/weather.models';

const BASE = '/weather-api';

@Injectable({ providedIn: 'root' })
export class WeatherService {
  private http = inject(HttpClient);

  getEntropyCurrent(postalCode = '93650', countryCode = 'US'): Observable<EntropyReading> {
    const params = new HttpParams()
      .set('postalCode', postalCode)
      .set('countryCode', countryCode);
    return this.http.get<EntropyReading>(`${BASE}/api/entropy/current`, { params });
  }

  getEntropyHistory(): Observable<EntropyHistory> {
    return this.http.get<EntropyHistory>(`${BASE}/api/entropy/history`);
  }
}

File: Services/WeatherService.cs
using System.Net.Http.Json;
using AgronomyStudio.Models;

namespace AgronomyStudio.Services;

public sealed class WeatherService
{
    private readonly HttpClient _http;

    public WeatherService(IHttpClientFactory factory)
    {
        _http = factory.CreateClient(ApiClients.Weather);
    }

    public Task<EntropyReading?> GetEntropyCurrentAsync(
        string postalCode = "93650",
        string countryCode = "US",
        CancellationToken ct = default) =>
        _http.GetFromJsonAsync<EntropyReading>(
            $"api/entropy/current?postalCode={Uri.EscapeDataString(postalCode)}&countryCode={Uri.EscapeDataString(countryCode)}",
            ct);

    public Task<EntropyHistory?> GetEntropyHistoryAsync(CancellationToken ct = default) =>
        _http.GetFromJsonAsync<EntropyHistory>("api/entropy/history", ct);
}
